test/utils.go

```diff
@@ -0,0 +1,66 @@
+package test
+
```

Comment:
Can't comment on file name in Github but perhaps we call this `tcp_test.go`?

a) this is only intended to be used in tests as far as I can see
b) being called utils we could put absolutely anything here, making it a bit more specific will allow better categorization of code that we put in here.

WDYT?

Reply:
Agree that we should make a more clear file name.

Reply:
changed to `tcp_utils.go`.